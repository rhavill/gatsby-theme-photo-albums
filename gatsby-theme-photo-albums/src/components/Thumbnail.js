import React from 'react'
import PropTypes from 'prop-types'
import {Link} from 'gatsby'
import Img from 'gatsby-image'
import {pathToFileTitle} from '../util/url-text'

const Thumbnail = ({path, imageData}) => {
  const title = pathToFileTitle(path)
  return <article className='file' data-testid='file'>
    <Link to={path}>
      <Img fixed={imageData} alt={title} title={title}/>
    </Link>
  </article>
}

Thumbnail.propTypes = {
  path: PropTypes.string.isRequired,
  imageData: PropTypes.shape({
    src: PropTypes.string.isRequired,
    srcSet: PropTypes.string.isRequired,
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
  }).isRequired  
}

export default Thumbnail