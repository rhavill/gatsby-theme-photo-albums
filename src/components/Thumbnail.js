import React from "react"
import PropTypes from "prop-types"
import {graphql, Link} from "gatsby"
import Img from "gatsby-image"

const Thumbnail = ({fileData}) =>  
  <article className='file'>
    <Link to={fileData.fsPath}>
      <Img fixed={fileData.fixed} />
    </Link>
  </article>
  
export const query = graphql`
  fragment ThumbnailFragment on FileConnection {
    edges {
      node {
        relativePath
        childImageSharp {
          fixed(width: 250, height: 250) {
            ...GatsbyImageSharpFixed
          }
        }
      }
    }
  }
`

Thumbnail.propTypes = {
  fileData: PropTypes.shape({
    fsPath: PropTypes.string.isRequired,
    fixed: PropTypes.shape({
      src: PropTypes.string.isRequired,
      srcSet: PropTypes.string.isRequired,
      width: PropTypes.number.isRequired,
      height: PropTypes.number.isRequired,
    }).isRequired
  }).isRequired  
}

export default Thumbnail