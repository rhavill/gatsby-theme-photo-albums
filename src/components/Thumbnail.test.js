import React from "react"
import renderer from "react-test-renderer"
import {cleanup, render} from '@testing-library/react';
import Thumbnail from "./Thumbnail"

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

const fileData = {
  fsPath: '/level-one/level-two/3.jpg',
  fixed: {
    width: 250,
    height: 250,
    src: '/static/hash/f17d3/folder.png',
    srcSet: '/static/hash/f17d3/folder.png 1x',
  }
}

describe("Thumbnail", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Thumbnail fileData={fileData} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
