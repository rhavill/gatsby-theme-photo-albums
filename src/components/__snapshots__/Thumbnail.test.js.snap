// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail renders correctly 1`] = `
<article
  className="file"
>
  <a
    href="/level-one/level-two/3.jpg"
  >
    <div
      className=" gatsby-image-wrapper"
      style={
        Object {
          "display": "inline-block",
          "height": 250,
          "overflow": "hidden",
          "position": "relative",
          "width": 250,
        }
      }
    >
      <picture>
        <source
          srcSet="/static/hash/f17d3/folder.png 1x"
        />
        <img
          alt=""
          height={250}
          loading="lazy"
          onLoad={[Function]}
          src="/static/hash/f17d3/folder.png"
          srcSet="/static/hash/f17d3/folder.png 1x"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
          width={250}
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><source srcset=\\"/static/hash/f17d3/folder.png 1x\\" /><img loading=\\"lazy\\" width=\\"250\\" height=\\"250\\" srcset=\\"/static/hash/f17d3/folder.png 1x\\" src=\\"/static/hash/f17d3/folder.png\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
  </a>
</article>
`;
